cmd/minikube/cmd/start.go

```diff
@@ -156,6 +169,36 @@ func runStart(cmd *cobra.Command, args []string) {
 	clusterUp(&config)
 }
 
+// Set docker and oc cluster proxy in place
+func setProxy() {
+	if viper.IsSet("http-proxy") {
+		dockerEnv = append(dockerEnv, fmt.Sprintf("HTTP_PROXY=%s", viper.GetString("http-proxy")))
+		clusterUpFlagSet.String(httpProxy, "", "HTTP proxy to use for master and builds")
+		viper.Set(httpProxy, viper.GetString("http-proxy"))
+		shellProxyEnv += fmt.Sprintf("http_proxy=%s ", viper.GetString("http-proxy"))
+	}
+	if viper.IsSet("https-proxy") {
+		dockerEnv = append(dockerEnv, fmt.Sprintf("HTTPS_PROXY=%s", viper.GetString("https-proxy")))
```

Comment:
Maybe what "bugs" me is that in this method you are setting up three things in one go - Docker, OpenShift and the env for the VM in general. That's quite intertwined and not really clear to the user. Opposed to something like setOpenShiftProxyEnv, setDockerProxyEnv, setVmProxyEnv.